### PR TITLE
Create data and data access object interfaces.

### LINKED ISSUE #5: Firebase SDK Dashboard: Directory Structure
Initialize the directory structure in the github repo

### LINKED ISSUE #7: Create data access object interfaces

### PR DESCRIPTION
**Pull request by eriyezab**
The data objects that I created are subject to change. One thing I realized is that I am not 100% certain on the fields each SDKRelease object should have since there seems to be a lot of columns in the spreadsheets that are optional and we also have not seen the spreadsheets or formats for all the different platforms. I was aiming to have all the mandatory attributes an SDKRelease object should have and then the rest of the fields that may be optional would go into a hashmap.

Close #7 

Create data access object(DAO) interfaces that will be extended and used to interact with the database as well as mock the database for testing. Allows for future database migrations without the need to modify api code.
DAO interfaces contain the names of the main methods that will be needed for the app.
Create data objects that represent the main features of the dashboard.

### CODE REVIEW

File: src/main/java/com/google/capstone/dao/PlatformReleaseDao.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package com.google.capstone.dao;
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/dao/PlatformReleaseDao.java:1]: Can you please update the directory to something like firebase/sdk/dashboard/dao/..?

capstone is specific to your internship project but cannot be extended.

File: src/main/java/com/google/capstone/data/Platform.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.google.capstone.data;
+
+import java.util.ArrayList;
+
+public class Platform {
+  private String name;
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/Platform.java:6]: Platform  could be just an enum as the values are fixed & will not change.

We can skip closestDeadline for now.

File: src/main/java/com/google/capstone/data/User.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.google.capstone.data;
+
+import java.util.ArrayList;
+
+public class User {
+  private long id;
+  private String email;
+  private ArrayList<SDK> favorites;
+
+  private User(long id, String email, ArrayList<SDK> favorites) {
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/User.java:10]: Use Builder design pattern for constructors & Autovalue library for getters/setters.

File: src/main/java/com/google/capstone/data/SDKRelease.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.util.HashMap;
+
+public class SDKRelease {
+  private String libraryName;
+  private String platform;
+  private String release;
+  private String newVersion;
+  private String oldVersion;
+  private boolean majorVersionBump;
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/SDKRelease.java:11]: not required right now. And there are 3 types of version bumps.

File: src/main/java/com/google/capstone/data/SDKRelease.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  /* This will contain optional information like links to CLs, ariane entries, etc., */
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/SDKRelease.java:13]: nit: remove the ,

File: src/main/java/com/google/capstone/data/SDK.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.LinkedHashMap;
+
+public class SDK {
+  private String libraryName;
+  private String libraryGroup;
+  private String externalName;
+  private String fireEscapeName;
+  private String owner;
+  private LinkedHashMap<String, SDKRelease> versionHistory;
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/SDK.java:11]: Why do we need an HashMap for this? Why can't it be a List<SDKRelease>? Also, versionHistory definition doesn't belong in this class. It should be part of Dao

**Comment by eriyezab** [comment]
I did not mean to close this pull request. Was working on the API branch and wanted to push some of the changes I made and it ended up merging and closing this branch. I did want to merge this branch because the API branch depends on this but I had not realized that doing so would close this pr @ankitaj224

**Review by ankitaj224 — COMMENTED** [review]
In general, DAO design pattern is not correctly implemented.  A simple tutorial: https://www.tutorialspoint.com/design_pattern/data_access_object_pattern.htm

No worries, if you merged it. But I would recommend raising another one fixing the comments & changing the model objects & DAOs.

File: src/main/java/com/google/capstone/data/User.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
+package com.google.capstone.data;
+
+import java.util.ArrayList;
+
+public class User {
+  private long id;
+  private String email;
+  private ArrayList<SDK> favorites;
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/User.java:8]: User class shouldn't contains List<SDK>, method addSDKToFavorites & removeSDKFromFavorites . This should be part of UserDao.

File: src/main/java/com/google/capstone/data/SDKRelease.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
+package com.google.capstone.data;
+
+import java.util.HashMap;
+
+public class SDKRelease {
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/SDKRelease.java:5]: SDKPerRelease or SDKReleaseMetadata

File: src/main/java/com/google/capstone/data/Release.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.ArrayList;
+
+public class Release {
+  private String platform;
+  private String releaseName;
+  private String releaseManager;
+  private long launchDeadline;
+  private long codeFreezeDate;
+  private long launchDate;
+  private ArrayList<String> enrolledSDKs;
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/Release.java:12]: part of dao.

File: src/main/java/com/google/capstone/data/Platform.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  private long closestDeadline;
+  
+  private ArrayList<Release> releases;
```
Comment (ankitaj224) [inline src/main/java/com/google/capstone/data/Platform.java:9]: part of dao

**Pull request merged**